src/RulesEngine/ParamCompiler.cs

```diff
@@ -122,26 +68,10 @@ private IEnumerable<ParameterExpression> GetParameterExpression(params RuleParam
         /// <param name="typeParameterExpressions">The type parameter expressions.</param>
         /// <param name="ruleInputExp">The rule input exp.</param>
         /// <returns></returns>
-        private Expression GetExpressionForRuleParam(LocalParam param, IEnumerable<ParameterExpression> typeParameterExpressions, ParameterExpression ruleInputExp)
-        {
-            return BuildExpression(param, typeParameterExpressions, ruleInputExp);
-        }
-
-        /// <summary>
-        /// Builds the expression.
-        /// </summary>
-        /// <param name="param">The rule.</param>
-        /// <param name="typeParameterExpressions">The type parameter expressions.</param>
-        /// <param name="ruleInputExp">The rule input exp.</param>
-        /// <returns></returns>
-        /// <exception cref="InvalidOperationException"></exception>
-        private Expression BuildExpression(LocalParam param, IEnumerable<ParameterExpression> typeParameterExpressions, ParameterExpression ruleInputExp)
+        private Delegate GetDelegateForRuleParam(LocalParam param, RuleParameter[] ruleParameters)
         {
-            var ruleExpressionBuilder = _expressionBuilderFactory.RuleGetExpressionBuilder(RuleExpressionType.LambdaExpression);
-
-            var expression = ruleExpressionBuilder.BuildExpressionForRuleParam(param, typeParameterExpressions, ruleInputExp);
-
-            return expression;
+            var expressionParser = new RuleExpressionParser(_reSettings);
```

Comment:
This will create a new instance of ruleExpressionParser. Need to ensure same instance is used.